analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/RestrictDeserializedTypesBase.cs

```diff
@@ -26,4 +29,169 @@ public abstract class RestrictDeserializedTypesBase : SymbolicRuleCheck
+        {
+            if (ProcessBinderAssignment(state, assignment) is { } binderProcessedState)
+            {
+                return binderProcessedState;
+            }
+            else if (AdditionalLocation(state, assignment.Value) is { } methodDeclaration
+                && assignment.Target.TrackedSymbol() is { } symbol)
+            {
+                additionalLocationsForSymbols[symbol] = methodDeclaration;
+            }
+        }
+        else if (UnsafeDeserialization(state, operation) is { } invocation)
+        {
+            var methodDeclaration = AdditionalLocation(state, invocation.Instance);
+            var additionalLocations = methodDeclaration is not null
+                ? new[] { GetIdentifier(methodDeclaration).GetLocation() }
+                : Array.Empty<Location>();
+            ReportIssue(operation, additionalLocations, RestrictTypesMessage);
+        }
+        return state;
+    }
+
+    private ProgramState ProcessOtherSerializerCreations(ProgramState state, IObjectCreationOperationWrapper objectCreation)
+    {
+        if (UnsafeJavaScriptSerializer(state, objectCreation, out var resolveTypeDeclaration))
+        {
+            additionalLocationsForOperations[objectCreation.WrappedOperation] = resolveTypeDeclaration;
+            return state.SetOperationConstraint(objectCreation.WrappedOperation, SerializationConstraint.Unsafe);
+        }
+        else if (UnsafeLosFormatter(state, objectCreation))
+        {
+            ReportIssue(objectCreation.WrappedOperation, VerifyMacMessage);
+        }
+        return state;
+    }
+
+    private bool UnsafeJavaScriptSerializer(ProgramState state, IObjectCreationOperationWrapper objectCreation, out SyntaxNode resolveTypeDeclaration)
+    {
+        resolveTypeDeclaration = null;
+        return objectCreation.Type.Is(JavaScriptSerializer)
+            && objectCreation.Arguments.Length == 1
+            && UnsafeResolver(state, objectCreation.Arguments[0].ToArgument().Value, out resolveTypeDeclaration);
+    }
+
+    private bool UnsafeResolver(ProgramState state, IOperation operation, out SyntaxNode resolveTypeDeclaration)
+    {
+        resolveTypeDeclaration = null;
+        if (state.ResolveCaptureAndUnwrapConversion(operation).Type.Is(KnownType.System_Web_Script_Serialization_SimpleTypeResolver))
+        {
+            return true;
+        }
+        else if (DeclarationCandidates(operation)?.FirstOrDefault(IsResolveType) is { } declaration
+            && !ThrowsOrReturnsNull(declaration))
+        {
+            resolveTypeDeclaration = declaration;
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    private static bool UnsafeLosFormatter(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        objectCreation.Type.Is(LosFormatter)
+        && objectCreation.ArgumentValue("enableMac") switch
+        {
+            null => true,
```

Comment:
I'd switch the order, null feels like the default part of the switch statement.
```cs
{ } enableMacArgument => state[state.ResolveCaptureAndUnwrapConversion(enableMacArgument)]?.HasConstraint(BoolConstraint.True) is not true,
_ => true,
```

Reply:
I completely rewrote it.